nixos/lib/systemd-lib.nix

```diff
@@ -21,11 +21,12 @@ in rec {
         { preferLocalBuild = true;
           allowSubstitutes = false;
           inherit (unit) text;
+          passAsFile = [ "text" ];
         }
         ''
           name=${shellEscape name}
           mkdir -p "$out/$(dirname -- "$name")"
-          echo -n "$text" > "$out/$name"
+          mv "$textPath" "$out/$name"
```

Comment:
Probably better to use `install` or `cp` rather than `mv`.

Reply:
See my review in https://github.com/NixOS/nixpkgs/pull/263555#discussion_r1373203448 why I suggested `mv` instead (if you haven't already)

Reply:
Hm, basically because `writeTextFile` does it wrong too? I mean, it's a super minor difference so I could easily be swayed either way. If everyone's happy with it this way then I'm totally on board.

Reply:
I don't have a strong opinion on this.
I just noticed `writeTextFile` does it and went with it.

This is your territory, and I am not part of the systemd team.
I just wasn't sure if you've seen my review suggestion when you wrote yours, that's all :)

Reply:
I just wanted to make sure you've considered it. If it's been fine for `writeTextFile` all this time, then it'll be fine here too. If you're happy with it, I'm cool with it too.